lua/toggleterm/ui.lua

```diff
@@ -10,10 +10,15 @@ local api = vim.api
 local persistent = {}
 local origin_window
 
-function M.save_window_size()
-  -- Save the size of the split before it is hidden
-  persistent.width = vim.fn.winwidth(0)
-  persistent.height = vim.fn.winheight(0)
+--- Save the size of a split window before it is hidden
+--- @param window number
+--- @param direction string
+function M.save_window_size(window, direction)
+  if direction == "horizontal" then
+    persistent.horizontal = vim.fn.winheight(window)
```

Comment:
This I think can actually be migrated to use api functions for a probably negligible speed increase e.g. `vim.api.nvim_win_get_{height|width}`